test/evp_extra_test.c

```diff
@@ -5230,6 +5234,16 @@ int setup_tests(void)
 {
     OPTION_CHOICE o;
 
+#ifdef STATIC_LEGACY
+    /*
+     * This test is always statically linked against libcrypto. We must not
+     * attempt to load legacy.so that might be dynamically linked against
+     * libcrypto. Instead we use a built-in version of the legacy provider.
+     */
+    if (!OSSL_PROVIDER_add_builtin(testctx, "legacy", ossl_legacy_provider_init))
+        return 0;
+#endif
+
```

Comment:
This needs to be moved below when the testctx is created but before `OSSL_PROVIDER_load(testctx, "legacy");` is called.

Reply:
> This needs to be moved below when the testctx is created but before `OSSL_PROVIDER_load(testctx, "legacy");` is called.

That fix works!